python/src/aac/lang/language_context.py

```diff
@@ -173,7 +176,7 @@ def remove_definition_from_context(self, definition: Definition):
         Args:
             definition (Definition): The Definition to remove from the context.
         """
-        if definition.uid in self.definitions_dictionary:
+        if definition.uid in self.definitions_dictionary and self._is_definition_editable(definition):
```

Comment:
I think we should make the `self._is_definition_editable(...)` call another `if` statement so we can give an error message that gives the user an appropriate understanding of what happened. Namely, if I try to delete `Primitives` I shouldn't be told that the definition isn't present but rather that the definition cannot be edited.

Reply:
Agreed & implemented